### PR TITLE
Prevent infinite load when avatar upload fails

### LINKED ISSUE #8877: Infinite loading issue while updating cover photo of a Facility
**Describe the bug**
When attempting to edit the cover image of a facility, if an image with an improper format is selected, the save button enters an indefinite "an uploading" state. The issue persists until the modal is reopened.

**To Reproduce**

1. Go to Facilty Tab
2. Select a Facilty
3. Click on **Upload** icon
4. Try uploading an image which is not in 1:1 format
5. Click Save and observe that it goes into loading state forever even after displaying the error

**Expected behavior**
If an upload is failed, the **save** button should regain it's regular state 

**Screen Recording**

https://github.com/user-attachments/assets/dee0b83e-e620-460d-9528-dc278827c22f

**Desktop (please complete the following information):**

- OS: Windows
- Browser : Chrome
- Version: Latest

### PR DESCRIPTION
**Pull request by nithish1018**
## Proposed Changes

- Fixes #8877 
- Handled failed upload

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### CODE REVIEW

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
image
image

When a user attempts to upload a non-permitted image, the system currently throws an error and closes the entire pop-up. This should not be the expected behavior. Instead, the user should remain on the same pop-up, where the error message is displayed, allowing them to re-select and upload an appropriate image.

**Comment by nithish1018** [comment]
> image image
> 
> When a user attempts to upload a non-permitted image, the system currently throws an error and closes the entire pop-up. This should not be the expected behavior. Instead, the user should remain on the same pop-up, where the error message is displayed, allowing them to re-select and upload an appropriate image.

Made some changes and this is how it works now:

https://github.com/user-attachments/assets/1ef8c9e7-8fd3-4f30-a315-6b8805c26947

File: src/Components/Common/AvatarEditModal.tsx [hunk 1]
```diff
@@ -114,11 +114,18 @@ const AvatarEditModal = ({
...
+        setPreview(undefined);
+        setPreviewImage(null);
+      });
+    } catch (e) {
+      console.error(e);
+    } finally {
       setIsCaptureImgBeingUploaded(false);
       setIsProcessing(false);
-    });
+    }
```
Comment (sainak) [inline src/Components/Common/AvatarEditModal.tsx]: These changes are not required, just calling onError() in the handler should be enough

Comment (nithishsiliveru) [reply src/Components/Common/AvatarEditModal.tsx]: Done, I removed them.
Is it ok now?

**Review by sainak — APPROVED** [review]

**Comment by rithviknishad** [comment]
it's still stuck in infinite loading :/

image

**Comment by nithish1018** [comment]
> it's still stuck in infinite loading :/
> 
> image

I checked it again and it's working properly for me 🤔

Uploading Recording 2024-10-23 162712.mp4…

**Comment by nithish1018** [comment]
> it's still stuck in infinite loading :/
> 
> image

I checked it again and it's working fine 🤔

https://github.com/user-attachments/assets/0455b33d-9e4e-4531-8479-9dc79ca58cba

**Comment by rithviknishad** [comment]
@nithish1018  can you try updating the profile pic instead of facility cover image?

**Comment by nithish1018** [comment]
> @nithish1018 can you try updating the profile pic instead of facility cover image?

Thank you, I added the fix for it

**Comment by rithviknishad** [comment]
clear merge conflicts; rest lgtm

**Review by rithviknishad — APPROVED** [review]

**Review by khavinshankar — APPROVED** [review]

**Comment by nihal467** [comment]
LGTM

**Pull request merged**